pkgs/development/tools/misc/blackfire/default.nix

```diff
@@ -8,37 +8,11 @@
 , common-updater-scripts
 }:
 
-let
-  version = "2.8.1";
-
-  sources = {
-    "x86_64-linux" = fetchurl {
-      url = "https://packages.blackfire.io/debian/pool/any/main/b/blackfire/blackfire_${version}_amd64.deb";
-      sha256 = "znaM00jM6yrpb+bGTxzJUxViCUzv4G+CYK2EB5dxhfY=";
-    };
-    "i686-linux" = fetchurl {
-      url = "https://packages.blackfire.io/debian/pool/any/main/b/blackfire/blackfire_${version}_i386.deb";
-      sha256 = "QIY4qGm333H5MWhe3CIfEieqTEk8st5A7SJHkwGnnxw=";
-    };
-    "aarch64-linux" = fetchurl {
-      url = "https://packages.blackfire.io/debian/pool/any/main/b/blackfire/blackfire_${version}_arm64.deb";
-      sha256 = "eZbKoKYC2tt4Rxn5OJr7iA1aJlYFC0tpRmbLq7qSrIU=";
-    };
-    "aarch64-darwin" = fetchurl {
-      url = "https://packages.blackfire.io/blackfire/${version}/blackfire-darwin_arm64.pkg.tar.gz";
-      sha256 = "tn2vF3v7KfF7CfWqyydL5Iyh5tP9Tez87PJH+URgSIw=";
-    };
-    "x86_64-darwin" = fetchurl {
-      url = "https://packages.blackfire.io/blackfire/${version}/blackfire-darwin_amd64.pkg.tar.gz";
-      sha256 = "CRFlnqpX4j2CMGzS+UvXwNty2mHpONOjym6UJPE2Yg4=";
-    };
-  };
-in
 stdenv.mkDerivation rec {
   pname = "blackfire";
-  inherit version;
+  version = "2.10.0";
 
-  src = sources.${stdenv.hostPlatform.system};
+  src = passthru.sources.${stdenv.hostPlatform.system};
```

Comment:
Missing or throw unsuppored system. Also first time seeing src in passthru

Reply:
It will fail implicitly and suggest a different system:

```ShellSession
$ nix-build -A blackfire --system aarch64-darwind
error: attribute 'aarch64-darwind' missing

       at /home/jtojnar/Projects/nixpkgs/pkgs/development/tools/misc/blackfire/default.nix:15:9:

           14|
           15|   src = passthru.sources.${stdenv.hostPlatform.system};
             |         ^
           16|
       Did you mean aarch64-darwin?

```

Reply:
> It will fail implicitly and suggest a different system:

That is not good enough because eval should not fail on other platforms.

> there is no point in having the package depend on all of the

nix is lazy so only the used src will be fetched.

Reply:
> Would not that be prevented by not evaluating on platforms not listed in `meta.platforms`?

but sometimes you want to evaluate (not build) the package on other platforms eg maintainer scripts or nixpkgs-review and that shouldn't fail.







> Not if you pass it to `mkDerivation` – then it will be forced to string in order so that it can be passed as environment variable to the builder.

oh, yeah for the updater script

Reply:
> but sometimes you want to evaluate (not build) the package on other platforms eg maintainer scripts or nixpkgs-review and that shouldn't fail.

Well, it will fail either way. The question is whether the default error is sufficient or if we need a more specific one.

Reply:
IIRC throw can be handled by nixpkgs-update through presumable tryEval unlike a syntax error.

Reply:
Not convinced this will help anything but I guess tge error is in fact slightly clearer:

```nix
nix-repl> builtins.tryEval blackfire
{ success = true; value = «derivation «error: error: Unsupported platform for blackfire: x86_64-linux»; }
```